### PR TITLE
CNV-3893 UI extended with Volume Mode and Access Mode for virtual disks

### PR DESCRIPTION
**Pull request by aburdenthehand**
The UI for virtual disks has been extended so users can specify Access Mode and Volume Mode. This has been added to the Wizard Storage Fields and Adding a Virtual Disk to a VM sections, with links to modifying the default settings in the ConfigMap.

[CNV-3893](https://issues.redhat.com/browse/CNV-3893)

### CODE REVIEW

**Comment by aburdenthehand** [comment]
Relevant build links:
https://cnv-3893-vm-disk-settings--ocpdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-edit-vms.html#virt-vm-add-disk_virt-edit-vms
https://cnv-3893-vm-disk-settings--ocpdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-create-vms.html#virt-storage-wizard-fields-web_virt-create-vms

File: modules/virt-storage-wizard-fields-web.adoc [hunk 1]
```diff
@@ -26,3 +26,36 @@
...
+These parameters are optional. If you do not specify these parameters, the defaults are used, based on what has been configured in the `kubevirt-storage-class-defaults` ConfigMap.
+
+[cols="2a,3a,5a",caption=]
+.Advanced storage settings
+|===
+|Name | Parameter |  Description
+
+.2+|Volume Mode
+|Filesystem
+|Creates a filesystem on the virtual disk.
```
Comment (aglitke) [inline modules/virt-storage-wizard-fields-web.adoc:43]: Stores the virtual disk in a file on a filesystem-based volume

File: modules/virt-storage-wizard-fields-web.adoc [hunk 2]
```diff
@@ -26,3 +26,36 @@
...
+
+|Block
+|Creates a block disk. Only use `Block` if the underlying storage supports it.
```
Comment (aglitke) [inline modules/virt-storage-wizard-fields-web.adoc:46]: Stores the virtual disk directly on the block volume.  Only use `Block` if the underlying storage supports it.

**Comment by gouyang** [comment]
Agree what @aglitke writes, others LGTM

File: modules/virt-add-disk-to-vm.adoc [hunk 3]
```diff
@@ -28,6 +28,7 @@ Use this procedure to add a virtual disk to a {object}.
 . Select the *Disks* tab.
 . Click *Add Disks* to open the *Add Disk* window.
 . In the *Add Disk* window, specify *Source*, *Name*, *Size*, *Interface*, and *Storage Class*.
+.. Optional: In the *Advanced* drop-down, specify the *Volume Mode* and *Access Mode* for the virtual disk. If you do not specify these parameters, the defaults are used, based on what has been configured in the `kubevirt-storage-class-defaults` ConfigMap.
```
Comment (ousleyp) [inline modules/virt-add-disk-to-vm.adoc:31]: s/`In the *Advanced* drop-down`/`From the *Advanced* list` (ISG page 209)

Nit: "the defaults are used, based on what has been configured in the" feels a bit wordy. What about something along these lines:

```
If you do not specify these parameters, the system uses the default values from the `kubevirt-storage-class-defaults` ConfigMap.
```

File: modules/virt-storage-wizard-fields-web.adoc [hunk 4]
```diff
@@ -26,3 +26,36 @@
 |The `StorageClass` that is used to create the disk.
 
 |===
+
+[id="virt-storage-wizard-fields-advanced-web_{context}"]
+[discrete]
+== Advanced storage settings
+The following advanced storage settings are available for *Blank*, *URL*, and *Attach Cloned Disk* disks.
+These parameters are optional. If you do not specify these parameters, the defaults are used, based on what has been configured in the `kubevirt-storage-class-defaults` ConfigMap.
```
Comment (ousleyp) [inline modules/virt-storage-wizard-fields-web.adoc:34]: Just noting that if you opt to change the wording of this sentence in the previous module, you should change it again here.

File: modules/virt-storage-wizard-fields-web.adoc [hunk 5]
```diff
@@ -26,3 +26,36 @@
...
+
+[cols="2a,3a,5a",caption=]
```
Comment (ousleyp) [inline modules/virt-storage-wizard-fields-web.adoc:36]: What is `caption=` for? I am unfamiliar.

**Review by ousleyp — COMMENTED** [review]
This is looking good - just a couple minor comments.

File: modules/virt-storage-wizard-fields-web.adoc (hunk 5 above)
Comment (aburdenthehand) [reply modules/virt-storage-wizard-fields-web.adoc:36]: When you give a table a header, by default it gets a 'Table [0-9]. ' prefix. This voids that prefix.
But I had since added a lvl2 section header and preceding paragraph, which made the table header redundant anyway so... it gone!
